app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,57 @@
+class RecipesController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @recipes = current_user.recipes.order(id: :asc)
+  end
+
+  def show
+    @recipe = Recipe.find(params[:id])
+    @recipe_foods = RecipeFood.where(recipe_id: params[:id])
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user = current_user
+    if @recipe.save
+      flash[:success] = 'recipe created successfully!'
+      redirect_to recipes_url
+    else
+      flash.now[:error] = 'Error: recipe could not be created!'
+      render :new, locals: { recipe: @recipe }
+    end
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id])
+    @recipe.destroy!
+    flash[:success] = 'Recipe was deleted successfully!'
+    redirect_to recipes_url
+  end
```

Comment:
- [ ] @Team it is good practice to create a success and fail paths when you create methods such as destroy, this enables easier troubleshooting in case of an error.